### PR TITLE
Pick sql index changes

### PR DESCRIPTION
**Pull request by yux0**
**What changed?**
1. Pick index change from https://github.com/temporalio/temporal/pull/1107
2. Fix go build with go 1.15.2 (https://github.com/golang/go/issues/40974)

**Why?**

**How did you test it?**

**Potential risks**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -127,23 +127,23 @@ copyright: cmd/tools/copyright/licensegen.go
 
 cadence-cassandra-tool: $(TOOLS_SRC)
 	@echo "compiling cadence-cassandra-tool with OS: $(GOOS), ARCH: $(GOARCH)"
-	go build -o cadence-cassandra-tool cmd/tools/cassandra/main.go
+	go build -ldflags '-w' -o cadence-cassandra-tool cmd/tools/cassandra/main.go
```
Comment (yycptt) [inline Makefile:130]: From that golang issue seems like this issue has been fixed in v1.15.4? Do we still need this change in this case?

File: schema/mysql/v57/cadence/versioned/v0.4/blob_size.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@ ALTER TABLE timer_info_maps MODIFY COLUMN data MEDIUMBLOB;
 ALTER TABLE child_execution_info_maps MODIFY COLUMN data MEDIUMBLOB;
 ALTER TABLE request_cancel_info_maps MODIFY COLUMN data MEDIUMBLOB;
 ALTER TABLE signal_info_maps MODIFY COLUMN data MEDIUMBLOB;
-ALTER TABLE buffered_replication_task_maps MODIFY COLUMN data MEDIUMBLOB;
+ALTER TABLE buffered_replication_task_maps MODIFY COLUMN new_run_history MEDIUMBLOB;
```
Comment (andrewjdawson2016) [inline schema/mysql/v57/cadence/versioned/v0.4/blob_size.sql:15]: Why are we renaming this column?

Comment (yux0) [reply schema/mysql/v57/cadence/versioned/v0.4/blob_size.sql:15]: This is the copy paste error from Temporal. Our column named this

File: Makefile (hunk 1 above)
Comment (yux0) [reply Makefile]: removed

**Review by yycptt — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -139,7 +139,7 @@ cadence: $(TOOLS_SRC)
 
 cadence-server: $(ALL_SRC)
 	@echo "compiling cadence-server with OS: $(GOOS), ARCH: $(GOARCH)"
-	go build -ldflags '$(GO_BUILD_LDFLAGS)' -o cadence-server cmd/server/main.go
+	go build -ldflags ' $(GO_BUILD_LDFLAGS)' -o cadence-server cmd/server/main.go
```
Comment (yycptt) [inline Makefile:142]: nit: extra space?

**Review by yycptt — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (yux0) [reply Makefile]: updated

**Pull request merged**